CMakeLists.txt

```diff
@@ -57,6 +56,23 @@ list(APPEND COMMON_COMPILE_OPTIONS
     $<$<CXX_COMPILER_ID:MSVC>:/EHsc>
 )
 
+
+################################################################################
+# Dependencies
+################################################################################
+add_subdirectory(third-party)
+
+# CMakes find_pacakge has no support for generator expressions hence we have to
+# this the hard way
+if(BUILD_CPPUNIT_TEST)
+    set(BOOST_COMPONENTS unit_test_framework)
+endif()
+
+set(Boost_USE_STATIC_LIBS ON)
+find_package(Boost 1.67 REQUIRED ${BOOST_COMPONENTS})
```

Comment:
is the hard-coded version here necessary?

Reply:
```cmake
find_package(Boost 1.67 REQUIRED ${BOOST_COMPONENTS})
```
means: Take a boost package that claims to be compatible to 1.67. What compatible means is up to the package. For boost that is usually to be interpreted as a min version. 

```cmake
find_package(Boost 1.67 EXACT REQUIRED ${BOOST_COMPONENTS})
```
would mean take ONLY Boost 1.67

I think I found a comment somewhere that 1.67 was the min version required but I am not entirely sure.

Reply:
Makes sense!